src/soundvectors.py

```diff
@@ -1,353 +1,439 @@
+    lab: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    hi: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    lo: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    back: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    front: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    tense: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    round: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    velaric: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    long: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    ant: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    distr: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    strid: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    hitone: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    hireg: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    loreg: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    rising: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    falling: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    contour: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    backshift: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    frontshift: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    opening: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    closing: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    centering: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    longdistance: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+    secondrounded: int = dataclasses.field(default=0, metadata={'dc:description': ''})
+
+    def __post_init__(self):
+        assert all(v in {0, 1, -1} for v in dataclasses.astuple(self))
+
+    @classmethod
+    def from_positive_and_negative(cls, pos: list, neg: list):
+        """
+        Factory method to allow for transparent instantiation.
+        """
+        assert not set(pos).intersection(neg)
+        return cls(**{k: 1 if k in pos else -1 for k in pos + neg})
+
+    def as_vector(self) -> typing.Tuple[int]:
+        return dataclasses.astuple(self)
 
-binary_features = ["cons", "syl", "son", "cont", "delrel", "lat", "nas", "voi", "sg", "cg", "pharyngeal", "laryngeal",
-    "cor", "dorsal", "lab", "hi", "lo", "back", "front", "tense", "round", "velaric", "long", "ant", "distr", "strid",
-    "hitone", "hireg", "loreg", "rising", "falling", "contour", "backshift", "frontshift", "opening", "closing",
-    "centering", "longdistance", "secondrounded", ]
```

Comment:
We don't need `binary_features` anymore because they can be inferred from `FeatureBundle` - including metadata.